contracts/src/utils/deployment/ProxyFactory.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity ^0.8.8;
+
+import {ProxyLib} from "./ProxyLib.sol";
+
+/// @title ProxyFactory
+/// @author Aragon Association - 2024
+/// @notice A factory to deploy proxies via the UUPS pattern (see [ERC-1822](https://eips.ethereum.org/EIPS/eip-1822)) and minimal proxy pattern (see [ERC-1167](https://eips.ethereum.org/EIPS/eip-1167)).
+/// @custom:security-contact sirt@aragon.org
+contract ProxyFactory {
```

Comment:
what is the idea of this contract?

Reply:
The idea of this contract is to provide a convenience contract for devs so that they can deploy initialized (but also uninitialized) UUPS and minimal proxies.

This is mainly for testing but can also be helpful for real deployments (although not particularly intended).
It might also be helpful for foundry devs.

The hardhat-upgrades package, which we used so far, allows you to do this only for UUPS proxies, but not with minimal proxies.
With this factory, you don't even need the hardhat-upgrades dependency in your project.

I also thought that having the same way of doing it for both proxies types is also convenient and reduces errors.